apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ast/builder/OperationTypeBuilder.kt

```diff
@@ -25,11 +27,14 @@ internal fun Operation.ast(
     isSubscription() -> OperationType.Type.SUBSCRIPTION
     else -> throw IllegalArgumentException("Unsupported GraphQL operation type: $operationType")
   }
+
+  val operationId = customIdGenerator?.apply(sourceWithFragments) ?: QueryDocumentMinifier.minify(sourceWithFragments).sha256()
```

Comment:
Don't we need to apply the `customIdGenerator` to the minified source ? This is what is going to be sent over the wire.

Also, we could introduce a `Sha256IdGenerator` so that this code ultimately becomes:

```kotlin
val operationId = (customIdGenerator ?: Sha256IdGenerator()).apply(QueryDocumentMinifier.minify(sourceWithFragments)) 
```

Reply:
I updated this change to use minified code to `apply`. Also, I add the Sha256IdGenerator.